applications/openshift/kubelet/kubelet_configure_tls_cipher_suites/ocp4/e2e-remediation.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+cat >> /etc/kubernetes/kubelet.conf << EOF
+tlsCipherSuites:
+- TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
+- TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
+- TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
+- TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
+EOF
```

Comment:
Note that the remediation does not run on the host itself. So... while this will run on the CI node, it doesn't really do anything but create this file on the test-runner. If you want to set this you'd need to set it via a custom `KubeletConfig` object: https://docs.openshift.com/container-platform/4.6/rest_api/machine_apis/kubeletconfig-machineconfiguration-openshift-io-v1.html or via `MachineConfig` directly

Reply:
Thank you, I misunderstood the behaviour of remediation run.

As far as I investigated, each of workaround has pros and cons.

### Using KubeletConfig
- document: [TLS configuration in OpenShift Container Platform \- Red Hat Customer Portal](https://access.redhat.com/articles/5348961)
- `.metadata.name` of `KubeletConfig` can be arbitrary defined (with specifying it as a label value for `machineconfigpool`), but names in check rule and remediation script must be predefined.  The check script may make false alert when user explicitly defines `KubeleteConfig` with a name which is not identical to the one in the check script.

### Using machineconfig
There are several `MachineConfig` resources in a CRC cluster.

```sh
$ oc get machineconfig
NAME                                               GENERATEDBYCONTROLLER                      IGNITIONVERSION   AGE
00-master                                          721141df0005d407d15b6664982f22be0604eb72   3.1.0             67d
00-worker                                          721141df0005d407d15b6664982f22be0604eb72   3.1.0             67d
01-master-container-runtime                        721141df0005d407d15b6664982f22be0604eb72   3.1.0             67d
01-master-kubelet                                  721141df0005d407d15b6664982f22be0604eb72   3.1.0             67d
01-worker-container-runtime                        721141df0005d407d15b6664982f22be0604eb72   3.1.0             67d
01-worker-kubelet                                  721141df0005d407d15b6664982f22be0604eb72   3.1.0             67d
...
```

In my understanding all of the `XX-XX-kubelet` `MachineConfig` resources should be checked.  However, [API Resource Collector of Compliance Operator](https://github.com/openshift/compliance-operator/blob/master/cmd/manager/api-resource-collector.go) does not support  fetching multiple resources (same issue happens [on authoring section 5 rules](https://github.com/ComplianceAsCode/content/issues/6529#issuecomment-773846031)), 

Any comments or suggestions? thanks in advance.

Reply:
@mrogers950 any idea what we could do here? ^^

Reply:
The remediation could still be a mc/KubeletConfig even though the check is looking at the /etc/kubernetes/kubelet.conf on the node. This way we're checking the running config of the kubelet as opposed to a MC that is only describing the intended state of the machines, so I think it makes for a more accurate check.

Reply:
I tried to apply KubeletConfig to confirm that applying KubeletConfig also changes `/etc/kubernetes/kubelet.conf`, but the method described in [this document](https://docs.openshift.com/container-platform/4.6/scalability_and_performance/recommended-host-practices.html#create-a-kubeletconfig-crd-to-edit-kubelet-parameters_) does not work on my CodeReady Container environment.  I pushed the remediation script to check the remediation script behaviour on CI test enviornment.